crates/uv-client/src/base_client.rs

```diff
@@ -461,6 +461,20 @@ impl RetryableStrategy for UvRetryableStrategy {
 ///
 /// These cases should be safe to retry with [`Retryable::Transient`].
 pub(crate) fn is_extended_transient_error(err: &dyn Error) -> bool {
+    // First, look for `WrappedReqwestError`, which wraps `reqwest::Error` but doesn't always
+    // include it in the source.
```

Comment:
We could also change `WrappedReqwestError`, it's a bit of a footgun, but I guess it's meant to _replace_ `reqwest::Error` rather than wrap it further.

Reply:
Yeah... idk. Both approaches seem to have downsides. I'd sort of rather fix the footgun but I'm not sure what other implications that has.

Reply:
`WrappedReqwestError` is a hack, if we have a better i'm all for it

Reply:
Yeah it looked sorta tricky. I can try though. I'll merge this for now.